xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -282,7 +282,7 @@ bool CGUIMediaWindow::OnMessage(CGUIMessage& message)
       else if (iControl == CONTROL_BTNSORTBY) // sort by
       {
         if (m_guiState.get())
-          m_guiState->SetNextSortMethod();
+          m_guiState->ChooseSortMethod();
```

Comment:
You don't use the result for anything. You should change this to return bool and only call UpdateFileList if the sort method was changed instead.